### PR TITLE
Cuda aware MPI is optional.

### LINKED ISSUE #212: G4 ring test is broken
This issue replaces #205.
It seems like the ring G accumulation test is fundamentally broken  as it fails on summit as well.
As a minor issue, I had to build on commit 0460fba9b80811781897e3311e582556cd754eff as PR #209 does not seem effective in automatically turning the test on on summit, and has other issues I mentioned in the PR.
```	
opening file : //gpfs/alpine/proj-shared/cph102/gbalduzz/DCA2/test/integration/cluster_solver/shared_tools/accumulation/tp/input_4x4_multitransfer.json


	 Parsing completed! read 1103 characters and 66 lines.
	 name        : CLUSTER MOMENTUM_SPACE BRILLOUIN_ZONE (DIMENSION : 2)
	 name (dual) : CLUSTER REAL_SPACE BRILLOUIN_ZONE (DIMENSION : 2)

	 size        : 4

			MOMENTUM_SPACE			|	REAL_SPACE
	 origin-index : 0				|	0
	 volume       : 3.947842e+01			|	4.000000e+00

	 basis : 
			3.141593e+00	-0.000000e+00	|	1.000000e+00	0.000000e+00	
			-0.000000e+00	3.141593e+00	|	0.000000e+00	1.000000e+00	

	 super-basis : 
			6.283185e+00	-0.000000e+00	|	2.000000e+00	0.000000e+00	
			-0.000000e+00	6.283185e+00	|	0.000000e+00	2.000000e+00	

	 inverse-basis : 
			3.183099e-01	0.000000e+00	|	1.000000e+00	-0.000000e+00	
			0.000000e+00	3.183099e-01	|	-0.000000e+00	1.000000e+00	

	 inverse-super-basis : 
			1.591549e-01	0.000000e+00	|	5.000000e-01	-0.000000e+00	
			0.000000e+00	1.591549e-01	|	-0.000000e+00	5.000000e-01	


	0	|	0.000000e+00	0.000000e+00		0.000000e+00	0.000000e+00	
	1	|	0.000000e+00	3.141593e+00		0.000000e+00	1.000000e+00	
	2	|	3.141593e+00	0.000000e+00		1.000000e+00	0.000000e+00	
	3	|	3.141593e+00	3.141593e+00		1.000000e+00	1.000000e+00	


	MOMENTUM_SPACE k-space symmetries : 

	0, 0	|		0, 0	0, 0	0, 0	0, 0	0, 0	0, 0	0, 0	0, 0
	0, 1	|		0, 1	0, 1	0, 1	0, 1	0, 1	0, 1	0, 1	0, 1
	1, 0	|		1, 0	2, 0	1, 0	2, 0	2, 0	1, 0	2, 0	1, 0
	1, 1	|		1, 1	2, 1	1, 1	2, 1	2, 1	1, 1	2, 1	1, 1
	2, 0	|		2, 0	1, 0	2, 0	1, 0	1, 0	2, 0	1, 0	2, 0
	2, 1	|		2, 1	1, 1	2, 1	1, 1	1, 1	2, 1	1, 1	2, 1
	3, 0	|		3, 0	3, 0	3, 0	3, 0	3, 0	3, 0	3, 0	3, 0
	3, 1	|		3, 1	3, 1	3, 1	3, 1	3, 1	3, 1	3, 1	3, 1



	REAL_SPACE symmetries : 

	0, 0	|		0, 0	0, 0	0, 0	0, 0	0, 0	0, 0	0, 0	0, 0
	0, 1	|		0, 1	0, 1	0, 1	0, 1	0, 1	0, 1	0, 1	0, 1
	1, 0	|		1, 0	2, 0	1, 0	2, 0	2, 0	1, 0	2, 0	1, 0
	1, 1	|		1, 1	2, 1	1, 1	2, 1	2, 1	1, 1	2, 1	1, 1
	2, 0	|		2, 0	1, 0	2, 0	1, 0	1, 0	2, 0	1, 0	2, 0
	2, 1	|		2, 1	1, 1	2, 1	1, 1	1, 1	2, 1	1, 1	2, 1
	3, 0	|		3, 0	3, 0	3, 0	3, 0	3, 0	3, 0	3, 0	3, 0
	3, 1	|		3, 1	3, 1	3, 1	3, 1	3, 1	3, 1	3, 1	3, 1

	 name        : LATTICE_SP MOMENTUM_SPACE BRILLOUIN_ZONE (DIMENSION : 2)
	 name (dual) : LATTICE_SP REAL_SPACE BRILLOUIN_ZONE (DIMENSION : 2)

	 size        : 1

			MOMENTUM_SPACE			|	REAL_SPACE
	 origin-index : 0				|	0
	 volume       : 3.947842e+01			|	1.000000e+00

	 basis : 
			6.283185e+00	-0.000000e+00	|	1.000000e+00	0.000000e+00	
			-0.000000e+00	6.283185e+00	|	0.000000e+00	1.000000e+00	

	 super-basis : 
			6.283185e+00	-0.000000e+00	|	1.000000e+00	0.000000e+00	
			-0.000000e+00	6.283185e+00	|	0.000000e+00	1.000000e+00	

	 inverse-basis : 
			1.591549e-01	0.000000e+00	|	1.000000e+00	-0.000000e+00	
			0.000000e+00	1.591549e-01	|	-0.000000e+00	1.000000e+00	

	 inverse-super-basis : 
			1.591549e-01	0.000000e+00	|	1.000000e+00	-0.000000e+00	
			0.000000e+00	1.591549e-01	|	-0.000000e+00	1.000000e+00	


	 name        : LATTICE_TP MOMENTUM_SPACE BRILLOUIN_ZONE (DIMENSION : 2)
	 name (dual) : LATTICE_TP REAL_SPACE BRILLOUIN_ZONE (DIMENSION : 2)

	 size        : 4

			MOMENTUM_SPACE			|	REAL_SPACE
	 origin-index : 0				|	0
	 volume       : 3.947842e+01			|	4.000000e+00

	 basis : 
			3.141593e+00	-0.000000e+00	|	1.000000e+00	0.000000e+00	
			-0.000000e+00	3.141593e+00	|	0.000000e+00	1.000000e+00	

	 super-basis : 
			6.283185e+00	-0.000000e+00	|	2.000000e+00	0.000000e+00	
			-0.000000e+00	6.283185e+00	|	0.000000e+00	2.000000e+00	

	 inverse-basis : 
			3.183099e-01	0.000000e+00	|	1.000000e+00	-0.000000e+00	
			0.000000e+00	3.183099e-01	|	-0.000000e+00	1.000000e+00	

	 inverse-super-basis : 
			1.591549e-01	0.000000e+00	|	5.000000e-01	-0.000000e+00	
			0.000000e+00	1.591549e-01	|	-0.000000e+00	5.000000e-01	


H_0 and H_int initialization start:    03-08-2020 09:50:20
H_0 and H_int initialization end:      03-08-2020 09:50:20
H_0 and H_int initialization duration: 6.509700e-05 s

G_0 initialization start:    03-08-2020 09:50:20
G_0 initialization end:      03-08-2020 09:50:20
G_0 initialization duration: 1.370800e-05 s

[1596462621.767523] [g22n11:24548:0]        knem_md.c:126  UCX  ERROR KNEM create region failed: Bad address
[1596462621.767571] [g22n11:24548:0]         ucp_mm.c:116  UCX  ERROR failed to register address 0x2000a3100400 length 100352 on md[11]=knem: Input/output error
[1596462621.767581] [g22n11:24548:0]    ucp_request.c:268  UCX  ERROR failed to register user buffer datatype 0x80 address 0x2000a3100400 len 100352: Input/output error
[g22n11:24548] ../../../../../../opensrc/ompi/ompi/mca/pml/ucx/pml_ucx.c:775  Error: ucx send failed: Input/output error
[g22n11:24548] *** An error occurred in MPI_Isend
[g22n11:24548] *** reported by process [564134428,1]
[g22n11:24548] *** on communicator MPI_COMM_WORLD
[g22n11:24548] *** MPI_ERR_OTHER: known error not in list
[g22n11:24548] *** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
[g22n11:24548] ***    and potentially your MPI job)
```

### PR DESCRIPTION
**Pull request by gbalduzz**
Edit: fixes #212
Solves manually #210.
~Depends on #206 or #208~

- Reverts the detection of multiple gpus and use a cmake flag instead for using CUDA aware MPI.
- Implements a (not really optimized) fallback for the ring algorithm that should make testing easier.

It seems that using the cvdlauncher script is not really necessary, and just using the launch flag `--smpiargs=”-gpu”` does the job on summit.

### CODE REVIEW

File: cmake/dca_testing.cmake [hunk 1]
```diff
@@ -131,13 +118,13 @@ function(dca_add_gtest name)
...
              COMMAND ${TEST_RUNNER} ${MPIEXEC_NUMPROC_FLAG} ${DCA_ADD_GTEST_MPI_NUMPROC}
-                     ${MPIEXEC_PREFLAGS} ${SMPIARGS_FLAG_MPI_CVD} ${CVD_LAUNCHER} "$<TARGET_FILE:${name}>")
+                     ${MPIEXEC_PREFLAGS} "$<TARGET_FILE:${name}>")
                  target_link_libraries(${name} ${MPI_C_LIBRARIES})
   else()
     if (TEST_RUNNER)
       add_test(NAME ${name}
                COMMAND ${TEST_RUNNER} ${MPIEXEC_NUMPROC_FLAG} 1
-	               ${MPIEXEC_PREFLAGS} ${SMPIARGS_FLAG_NOMPI} "$<TARGET_FILE:${name}>")
+	               ${MPIEXEC_PREFLAGS} "$<TARGET_FILE:${name}>")
```
Comment (gbalduzz) [inline cmake/dca_testing.cmake:127]: @PDoakORNL: I am not really aware of what the whole *_CVD flags where supposed to do or when they where introduced, so if I am missing something by removing them, please let me know.

File: build-aux/summit.cmake [hunk 2]
```diff
@@ -23,14 +23,11 @@ set(MPIEXEC_PREFLAGS "-a 1 -g 1 -c 5" CACHE STRING
 set(SMPIARGS_FLAG_NOMPI "--smpiargs=none" CACHE STRING
   "Spectrum MPI argument list flag for serial tests.")
 # Let's keep this option in case we need it again in the future.
-set(SMPIARGS_FLAG_MPI "" CACHE STRING "Spectrum MPI argument list flag for MPI tests.")
-
-# When we want to us a cuda visible devices restriction we need this flag
-set(SMPIARGS_FLAG_MPI_CVD "--smpiargs=-gpu" CACHE STRING
```
Comment (PDoakORNL) [inline build-aux/summit.cmake:29]: Change the name to _MGPU or something and don't remove it.  I have to build and test the code on more systems than a laptop and summit and its is useful to be able to partition the multigpu tests.  This is a useful distinction as long as our tests target 1 node.

File: CMakeLists.txt [hunk 3]
```diff
@@ -147,11 +147,7 @@ set(DCA_LIBS
   cuda_utils
 )
 
-set(SYSTEM_GPU_COUNT 0)
-
 if (DCA_HAVE_CUDA)
-  EXECUTE_PROCESS(COMMAND bash -c "nvidia-smi -L | awk 'BEGIN { num_gpu=0;} /GPU/ { num_gpu++;} END { printf(\"%d\", num_gpu) }'"
```
Comment (PDoakORNL) [inline CMakeLists.txt:153]: This is quire useful to have to make decisions about which tests to add.

**Comment by gbalduzz** [comment]
@PDoakORNL the ringG test does not use multiple GPUs: each rank has its own GPU. Currently DCA does not support multi-GPU.

File: cmake/dca_testing.cmake (hunk 1 above)
Comment (gbalduzz) [reply cmake/dca_testing.cmake]: Readded the cvdlauncher and a cache flag that can point to it (to be set from the <system>.cmake file) I kept the CUDA_CVD option out of the test function as all the tests have the same requirements besides requiring CUDA and/or MPI.

File: build-aux/summit.cmake (hunk 2 above)
Comment (gbalduzz) [reply build-aux/summit.cmake:29]: the additional flags can be easily added to the SMPIARGS_FLAG_MPI, as they don't need to differ between tests, even if the mpi implementation is not spectrum (sloppy naming on our side).

File: CMakeLists.txt (hunk 3 above)
Comment (gbalduzz) [reply CMakeLists.txt:153]: no test  depends on  the number of GPUs.

**Comment by PDoakORNL** [comment]
I'm making some additional changes to this PR based on testing on no summit multi GPU per node systems which I will PR back to you @gbalduzz.  This should hopefully mean we can meet our mpi/gpu testing needs without more changes at least for a couple months.

**Comment by gbalduzz** [comment]
Merged master.
Note that the ringG test is an integration test, so it requires the DCA_WITH_TESTS_EXTENSIVE=ON flag for being built. I would have preferred it as a unit test, but it does not really matter, and I did not want to change the original placement of the test.

**Comment by weilewei** [comment]
The ringG test jsrun arguments are not correct on Summit
```
cmake -C ../build-aux/summit.cmake -DDCA_WITH_TESTS_EXTENSIVE=On ..
make ringG_tp_accumulator_gpu_test
cd test/integration/cluster_solver/shared_tools/accumulation/tp/
```
The jsrun command line generated from ctest is (which causes hang and does not picks up GPUDirect on Summit)
```
bash-4.2$ ctest -V -N
....
1: Test command: /sw/summit/xalt/1.2.0/bin/jsrun "-n" "3" "-a" "1" "-g" "1" "-c" "5" "/gpfs/alpine/proj-shared/cph102/weile/dev/src/dca_giovanni/DCA/build_test/test/integration/cluster_solver/shared_tools/accumulation/tp/ringG_tp_accumulator_gpu_test"
  Test #1: ringG_tp_accumulator_gpu_test
```
The correct command line should be (which passed): 
```
jsrun -n3 -a1 -c7 -g1 -b rs --smpiargs="-gpu" ./cvdlauncher.sh ./ringG_tp_accumulator_gpu_test
```
So two command options are missing in ctest generated commands: ` --smpiargs="-gpu" ./cvdlauncher.sh `

Can you add these two to cmake related files? Or am I missing any settings before running distG4 related test? Otherwise, the rest of changes look good to me. Thanks.

**Comment by gbalduzz** [comment]
The cvd launcher is not necessary on summit: there is already one gpu per rank: the test passes as it is:
```
bash-4.2$ ctest -V   
UpdateCTestConfiguration  from :/gpfs/alpine/proj-shared/cph102/gbalduzz/DCA/build2/test/integration/cluster_solver/shared_tools/accumulation/tp/DartConfiguration.tcl
UpdateCTestConfiguration  from :/gpfs/alpine/proj-shared/cph102/gbalduzz/DCA/build2/test/integration/cluster_solver/shared_tools/accumulation/tp/DartConfiguration.tcl
Test project /gpfs/alpine/proj-shared/cph102/gbalduzz/DCA/build2/test/integration/cluster_solver/shared_tools/accumulation/tp
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 1
    Start 1: ringG_tp_accumulator_gpu_test

1: Test command: /sw/summit/xalt/1.2.0/bin/jsrun "-n" "3" "-a" "1" "-g" "1" "-c" "5" "//gpfs/alpine/proj-shared/cph102/gbalduzz/DCA/build2/test/integration/cluster_solver/shared_tools/accumulation/tp/ringG_tp_accumulator_gpu_test"
1: Test timeout computed to be: 10000000
1: Running main() from gtest_main.cc
1: [==========] Running 1 test from 1 test case.
1: [----------] Global test environment set-up.
1: [----------] 1 test from DistributedTpAccumulatorGpuTest
1: [ RUN      ] DistributedTpAccumulatorGpuTest.Accumulate
1: Running main() from gtest_main.cc
1: [==========] Running 1 test from 1 test case.
1: [----------] Global test environment set-up.
1: [----------] 1 test from DistributedTpAccumulatorGpuTest
1: [ RUN      ] DistributedTpAccumulatorGpuTest.Accumulate
1: Running main() from gtest_main.cc
1: [==========] Running 1 test from 1 test case.
1: [----------] Global test environment set-up.
1: [----------] 1 test from DistributedTpAccumulatorGpuTest
1: [ RUN      ] DistributedTpAccumulatorGpuTest.Accumulate
1: 
1: 
1: 	opening file : //gpfs/alpine/proj-shared/cph102/gbalduzz/DCA/test/integration/cluster_solver/shared_tools/accumulation/tp/input_4x4_multitransfer.json
1: 
1: 
1: 	 Parsing completed! read 1103 characters and 66 lines.
1: 	 name        : CLUSTER MOMENTUM_SPACE BRILLOUIN_ZONE (DIMENSION : 2)
1: 	 name (dual) : CLUSTER REAL_SPACE BRILLOUIN_ZONE (DIMENSION : 2)
1: 
1: 	 size        : 4
1: 
1: 			MOMENTUM_SPACE			|	REAL_SPACE
1: 	 origin-index : 0				|	0
1: 	 volume       : 3.947842e+01			|	4.000000e+00
1: 
1: 	 basis : 
1: 			3.141593e+00	-0.000000e+00	|	1.000000e+00	0.000000e+00	
1: 			-0.000000e+00	3.141593e+00	|	0.000000e+00	1.000000e+00	
1: 
1: 	 super-basis : 
1: 			6.283185e+00	-0.000000e+00	|	2.000000e+00	0.000000e+00	
1: 			-0.000000e+00	6.283185e+00	|	0.000000e+00	2.000000e+00	
1: 
1: 	 inverse-basis : 
1: 			3.183099e-01	0.000000e+00	|	1.000000e+00	-0.000000e+00	
1: 			0.000000e+00	3.183099e-01	|	-0.000000e+00	1.000000e+00	
1: 
1: 	 inverse-super-basis : 
1: 			1.591549e-01	0.000000e+00	|	5.000000e-01	-0.000000e+00	
1: 			0.000000e+00	1.591549e-01	|	-0.000000e+00	5.000000e-01	
1: 
1: 
1: 	0	|	0.000000e+00	0.000000e+00		0.000000e+00	0.000000e+00	
1: 	1	|	0.000000e+00	3.141593e+00		0.000000e+00	1.000000e+00	
1: 	2	|	3.141593e+00	0.000000e+00		1.000000e+00	0.000000e+00	
1: 	3	|	3.141593e+00	3.141593e+00		1.000000e+00	1.000000e+00	
1: 
1: 
1: 	MOMENTUM_SPACE k-space symmetries : 
1: 
1: 	0, 0	|		0, 0	0, 0	0, 0	0, 0	0, 0	0, 0	0, 0	0, 0
1: 	0, 1	|		0, 1	0, 1	0, 1	0, 1	0, 1	0, 1	0, 1	0, 1
1: 	1, 0	|		1, 0	2, 0	1, 0	2, 0	2, 0	1, 0	2, 0	1, 0
1: 	1, 1	|		1, 1	2, 1	1, 1	2, 1	2, 1	1, 1	2, 1	1, 1
1: 	2, 0	|		2, 0	1, 0	2, 0	1, 0	1, 0	2, 0	1, 0	2, 0
1: 	2, 1	|		2, 1	1, 1	2, 1	1, 1	1, 1	2, 1	1, 1	2, 1
1: 	3, 0	|		3, 0	3, 0	3, 0	3, 0	3, 0	3, 0	3, 0	3, 0
1: 	3, 1	|		3, 1	3, 1	3, 1	3, 1	3, 1	3, 1	3, 1	3, 1
1: 
1: 
1: 
1: 	REAL_SPACE symmetries : 
1: 
1: 	0, 0	|		0, 0	0, 0	0, 0	0, 0	0, 0	0, 0	0, 0	0, 0
1: 	0, 1	|		0, 1	0, 1	0, 1	0, 1	0, 1	0, 1	0, 1	0, 1
1: 	1, 0	|		1, 0	2, 0	1, 0	2, 0	2, 0	1, 0	2, 0	1, 0
1: 	1, 1	|		1, 1	2, 1	1, 1	2, 1	2, 1	1, 1	2, 1	1, 1
1: 	2, 0	|		2, 0	1, 0	2, 0	1, 0	1, 0	2, 0	1, 0	2, 0
1: 	2, 1	|		2, 1	1, 1	2, 1	1, 1	1, 1	2, 1	1, 1	2, 1
1: 	3, 0	|		3, 0	3, 0	3, 0	3, 0	3, 0	3, 0	3, 0	3, 0
1: 	3, 1	|		3, 1	3, 1	3, 1	3, 1	3, 1	3, 1	3, 1	3, 1
1: 
1: 	 name        : LATTICE_SP MOMENTUM_SPACE BRILLOUIN_ZONE (DIMENSION : 2)
1: 	 name (dual) : LATTICE_SP REAL_SPACE BRILLOUIN_ZONE (DIMENSION : 2)
1: 
1: 	 size        : 1
1: 
1: 			MOMENTUM_SPACE			|	REAL_SPACE
1: 	 origin-index : 0				|	0
1: 	 volume       : 3.947842e+01			|	1.000000e+00
1: 
1: 	 basis : 
1: 			6.283185e+00	-0.000000e+00	|	1.000000e+00	0.000000e+00	
1: 			-0.000000e+00	6.283185e+00	|	0.000000e+00	1.000000e+00	
1: 
1: 	 super-basis : 
1: 			6.283185e+00	-0.000000e+00	|	1.000000e+00	0.000000e+00	
1: 			-0.000000e+00	6.283185e+00	|	0.000000e+00	1.000000e+00	
1: 
1: 	 inverse-basis : 
1: 			1.591549e-01	0.000000e+00	|	1.000000e+00	-0.000000e+00	
1: 			0.000000e+00	1.591549e-01	|	-0.000000e+00	1.000000e+00	
1: 
1: 	 inverse-super-basis : 
1: 			1.591549e-01	0.000000e+00	|	1.000000e+00	-0.000000e+00	
1: 			0.000000e+00	1.591549e-01	|	-0.000000e+00	1.000000e+00	
1: 
1: 
1: 	 name        : LATTICE_TP MOMENTUM_SPACE BRILLOUIN_ZONE (DIMENSION : 2)
1: 	 name (dual) : LATTICE_TP REAL_SPACE BRILLOUIN_ZONE (DIMENSION : 2)
1: 
1: 	 size        : 4
1: 
1: 			MOMENTUM_SPACE			|	REAL_SPACE
1: 	 origin-index : 0				|	0
1: 	 volume       : 3.947842e+01			|	4.000000e+00
1: 
1: 	 basis : 
1: 			3.141593e+00	-0.000000e+00	|	1.000000e+00	0.000000e+00	
1: 			-0.000000e+00	3.141593e+00	|	0.000000e+00	1.000000e+00	
1: 
1: 	 super-basis : 
1: 			6.283185e+00	-0.000000e+00	|	2.000000e+00	0.000000e+00	
1: 			-0.000000e+00	6.283185e+00	|	0.000000e+00	2.000000e+00	
1: 
1: 	 inverse-basis : 
1: 			3.183099e-01	0.000000e+00	|	1.000000e+00	-0.000000e+00	
1: 			0.000000e+00	3.183099e-01	|	-0.000000e+00	1.000000e+00	
1: 
1: 	 inverse-super-basis : 
1: 			1.591549e-01	0.000000e+00	|	5.000000e-01	-0.000000e+00	
1: 			0.000000e+00	1.591549e-01	|	-0.000000e+00	5.000000e-01	
1: 
1: 
1: H_0 and H_int initialization start:    26-08-2020 14:39:47
1: H_0 and H_int initialization end:      26-08-2020 14:39:47
1: H_0 and H_int initialization duration: 7.881000e-05 s
1: 
1: G_0 initialization start:    26-08-2020 14:39:47
1: G_0 initialization end:      26-08-2020 14:39:47
1: G_0 initialization duration: 1.495300e-05 s
1: 
1: [       OK ] DistributedTpAccumulatorGpuTest.Accumulate (2644 ms)
1: [----------] 1 test from DistributedTpAccumulatorGpuTest (2644 ms total)
1: 
1: [----------] Global test environment tear-down
1: [==========] 1 test from 1 test case ran. (2644 ms total)
1: [  PASSED  ] 1 test.
1: [       OK ] DistributedTpAccumulatorGpuTest.Accumulate (2644 ms)
1: [----------] 1 test from DistributedTpAccumulatorGpuTest (2644 ms total)
1: 
1: [----------] Global test environment tear-down
1: [==========] 1 test from 1 test case ran. (2644 ms total)
1: [  PASSED  ] 1 test.
1: [       OK ] DistributedTpAccumulatorGpuTest.Accumulate (2644 ms)
1: [----------] 1 test from DistributedTpAccumulatorGpuTest (2644 ms total)
1: 
1: [----------] Global test environment tear-down
1: [==========] 1 test from 1 test case ran. (2644 ms total)
1: [  PASSED  ] 1 test.
1/1 Test #1: ringG_tp_accumulator_gpu_test ....   Passed    4.63 sec
```

**Comment by weilewei** [comment]
Ok, I re-run it again, it works now (not sure why previous run fails). LGTM, thanks!

**Pull request merged**